daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Command.scala

```diff
@@ -15,30 +15,79 @@ sealed abstract class Command extends Product with Serializable {
 
 object Command {
 
+  /** Create a template, not by interface */
   final case class Create(
       templateId: Identifier,
       argument: SValue,
   ) extends Command
 
+  /** Create a template, by interface */
+  final case class CreateByInterface(
+      interfaceId: Identifier,
+      templateId: Identifier,
+      argument: SValue,
+  ) extends Command
+
+  /** Exercise a template choice, not by interface */
   final case class Exercise(
       templateId: Identifier,
       contractId: SContractId,
       choiceId: ChoiceName,
       argument: SValue,
   ) extends Command
 
+  /** Exercise a template choice, by interface */
+  final case class ExerciseByInterface(
+      interfaceId: Identifier,
+      templateId: Identifier,
+      contractId: SContractId,
+      choiceId: ChoiceName,
+      argument: SValue,
+  ) extends Command
+
+  /** Exercise an interface choice */
```

Comment:
```suggestion
  /** Exercise an interface choice by template*/
```
maybe? if not I think I’m confused

Reply:
Not by template. It exercises an interface without knowing the template id. This is what you get when exercising an interface via the ledger api, e.g.

Reply:
Ah I see, maybe make it explicit that this is used via the ledger API.

Reply:
Added a comment.